src/main/kotlin/de/triplet/gradle/play/PlayPublishListingTask.kt

```diff
@@ -0,0 +1,166 @@
+                            .setContactPhone(phone)
+                            .setContactWebsite(web)
+                            .setDefaultLanguage(defaultLanguage)
+
+                    edits.details()
+                            .update(variant.applicationId, editId, details)
+                            .execute()
+                }
+
+
+                edits.commit(variant.applicationId, editId).execute()
+            }
+        }
+    }
+
+    private fun updateListing(dir: File) {
+        val locale = dir.name
+
+        val listingDir = File(dir, LISTING_PATH)
+        // Check if listing directory exist
+        if (listingDir.exists()) {
+            val listing = Listing()
+
+            val fileTitle = File(listingDir, FILE_NAME_FOR_TITLE)
+            val title = fileTitle.readAndTrim(MAX_CHARACTER_LENGTH_FOR_TITLE, extension.errorOnSizeLimit)
+            if (!title.isEmpty()) {
+                listing.title = title
+            }
+
+            val fileShortDescription = File(listingDir, FILE_NAME_FOR_SHORT_DESCRIPTION)
+            val shortDescription = fileShortDescription.readAndTrim(MAX_CHARACTER_LENGTH_FOR_SHORT_DESCRIPTION, extension.errorOnSizeLimit)
+            if (!shortDescription.isEmpty()) {
+                listing.shortDescription = shortDescription
+            }
+
+            val fileFullDescription = File(listingDir, FILE_NAME_FOR_FULL_DESCRIPTION)
+            val fullDescription = fileFullDescription.readAndTrim(MAX_CHARACTER_LENGTH_FOR_FULL_DESCRIPTION, extension.errorOnSizeLimit)
+            if (!fullDescription.isEmpty()) {
+                listing.fullDescription = fullDescription
+            }
+
+            val video = File(listingDir, FILE_NAME_FOR_VIDEO).firstLine()
+            if (!video.isEmpty()) {
+                listing.video = video
+            }
+
+            try {
+                edits.listings()
+                        .update(variant.applicationId, editId, locale, listing)
+                        .execute()
+            } catch (e: GoogleJsonResponseException) {
+                // In case we are using an unsupported locale Google generates an error that
+                // is not exactly helpful. In that case we just wrap the original exception in our own.
+                if (e.message != null && e.message.toString().contains("unsupportedListingLanguage")) {
+                    throw IllegalArgumentException("Unsupported locale $locale", e)
+                }
+
+                // Just rethrow everything else.
+                throw e
+            }
+
+            // By default this will be skipped – can be enabled in the play extension
+            if (extension.uploadImages) {
+                // Only one ContentFile allowed for featureGraphic
+                val featureGraphicContent = getImages(listingDir, "$IMAGE_TYPE_FEATURE_GRAPHIC/")
+                uploadImages(featureGraphicContent, locale, IMAGE_TYPE_FEATURE_GRAPHIC, 1)
```

Comment:
We can (should?) however change this into an error. In the same change we should move the validations ahead of any executions of the Play Publisher API, so that the executions/updates are "atomic". See #167 as a possible integration.